dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -446,7 +513,10 @@ case class BinaryOutcomeDLCClient(
       .addUTXOToInput(fundingTx, index = 0)
       .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
       .sign(inputIndex = 0, fundingPrivKey)
-      .map(_.inputMaps.head.partialSignatures.head)
+      .map(
+        _.inputMaps.head.partialSignatures
+          .find(_.pubKey == fundingPrivKey.publicKey)
+          .get)
```

Comment:
same comment as above, if this pattern is re-used in more places maybe just factor it out into a helper method?